packages/create-app/src/index.ts

```diff
@@ -0,0 +1,410 @@
+  const pkgManager = pkgInfo ? pkgInfo.name : 'npm'
+
+  // Step 2. Copy the templates
+
+  console.log(`\nScaffolding templates in ${rootPath}...`)
+
+  for (const template of templates) {
+    const callerPath = fileURLToPath(import.meta.url)
+    const templateDir = `template-${template.name}`
+    const templateSrc = path.resolve(callerPath, '../..', templateDir)
+    const templateDest = path.join(rootPath, template.targetDir ?? `packages/${template.name}`)
+
+    // Exit if no template files found
+    if (!fs.existsSync(templateSrc)) {
+      continue
+    }
+
+    if (fs.existsSync(templateDest) && overwrite === 'yes') {
+      emptyDir(templateDest)
+    } else {
+      fs.mkdirSync(templateDest, { recursive: true })
+    }
+
+    const write = (file: string, content?: string) => {
+      const targetPath = path.join(templateDest, moveFiles[file] ?? file)
+      if (content) {
+        fs.writeFileSync(targetPath, content)
+      } else {
+        copy(path.join(templateSrc, file), targetPath)
+      }
+    }
+
+    const files = fs.readdirSync(templateSrc)
+    for (const file of files) {
+      write(file)
+    }
+  }
+
+  // Step 3. Run the custom commands
+
+  console.log(`\nDone. Now running template commands.`)
+
+  // Flatten all selected template commands
+  const customCommands = templates
+    .map((t) => t.customCommands)
+    .flat()
+    .filter(Boolean) as Command[]
+
+  // Run commands from project root
+  if (rootPath !== cwd) {
+    spawnSync('cd', [rootPath], { stdio: 'inherit' })
+  }
+
+  for (const customCommand of customCommands) {
+    // `node:fs` commands as are stricly typed arrays
+    if (Array.isArray(customCommand)) {
+      const [command, ...args] = customCommand
+      const formattedArgs = args.map((arg) =>
+        typeof arg === 'string' ? formatCommand(arg) : arg,
+      ) as typeof args
+      fs[command](...formattedArgs)
+      continue
+    }
+
+    // `spawnSync` commands are loosely typed strings
+    const formattedCommand = formatCommand(customCommand)
+    const [command, ...args] = splitCommand(formattedCommand) as [string, ...string[]]
+    spawnSync(command, args, { stdio: 'inherit' })
+  }
+
+  // Step 4. Install dependencies
+
+  console.log(`\nDone. Now installing dependencies.\n`)
+
+  if (pkgManager) {
+    spawnSync(pkgManager, ['install'], { stdio: 'inherit' })
+  }
+
+  // Step 5. Print goodbye message
+
+  const cdProjectName = path.relative(cwd, rootPath)
+  console.log(`\nDone. Now run:\n`)
+  if (rootPath !== cwd) {
+    console.log(`  cd ${cdProjectName.includes(' ') ? `"${cdProjectName}"` : cdProjectName}`)
+  }
+  console.log(`  ${pkgManager} run dev`)
+  console.log()
+}
+
+function formatTargetDir(targetDir: string | undefined) {
+  return targetDir?.trim().replace(/\/+$/g, '')
+}
+
+function formatCommand(command: string) {
+  const pkgInfo = pkgFromUserAgent(process.env.npm_config_user_agent)
+  const pkgManager = pkgInfo ? pkgInfo.name : 'npm'
```

Comment:
**question:** How come we fallback to `npm` here?

Have a hunch that we may provide better UX if we can print an error if we can't properly determine the package manager.

Maybe we can add the functionality to pass the package manager like turborepo (`-m`) as well?

Reply:
A lot from this PR was inspired by `vite-create` and that is how they handle checking for what package manager you used to run the executable. That said, the fallback is for those package managers that do not have support for  `npm_config_user_agent`. This is supported by the major package managers like `bun`, `npm`, `pnpm` & `yarn`.

As for the `-m` flag in turbo, this script handles it differently. The package manager you used at install time is the one that will be used internally for installation.